assets/src/scripts/components/hydrograph/index.js

```diff
@@ -43,23 +43,75 @@ class Hydrograph {
      * @param {Node} element Dom node to insert
      */
     constructor({data=[], yLabel='Data', title='', desc='', element}) {
-        this._data = data;
         this._yLabel = yLabel;
         this._title = title;
         this._desc = desc;
         this._element = element;
+        this._tsData = {};
 
-        if (this._data && this._data.length) {
+        if (data && data.length) {
+            this._tsData.current = data;
             this._drawChart();
         } else {
             this._drawMessage('No data is available for this site.');
         }
     }
 
+    /**
+     * Add a new time series to the Hydrograph. The time series is assumed to be
+     * data that is over the same date range in a different year.
+     * @param {Array} data - IV data as returned by models.getTimeseires
+     */
+    addCompareTimeSeries(data) {
+        //Save data - TODO will be needed in order to implement the tooltips
+        this._tsData.compare = data;
+
+        // Update the yScale by determining the new extent
+        const currentYExtent = extent(this._tsData.current, d => d.value);
+        const yExtent = extent(data, d => d.value);
+        const yDataExtent = [min([yExtent[0], currentYExtent[0]]), max([yExtent[1], currentYExtent[1]])];
+        updateYScale(this.scale.yScale, yDataExtent);
+
+        // Create a x scale for the new data
+        const xScale = createXScale(data, WIDTH - MARGIN.right);
+
+        // Update the yAxis
+        updateYAxis(this.axis.yAxis, this.scale.yScale);
+        this.svg.select('.y-axis')
+            .call(this.axis.yAxis);
+
+        //Update the current ts line
+        select('#ts-current')
+            .attr('d', this.currentLine(this._tsData.current));
+
+        // Add the new time series
+        this._plotDataLine(this.plot, {xScale: xScale, yScale: this.scale.yScale}, 'compare');
+    }
+
+    /**
+     * Remove the compare time series from the plot and rescale
+     */
+    removeCompareTimeSeries() {
+        // Remove the compare time series
+        this.svg.select('#ts-compare').remove();
+        this.svg.select('.x-top-axis').remove();
+        delete this._tsData.compare;
+
+        // Update the y scale and  redraw the axis
+        const currentYExtent = extent(this._tsData.current, d => d.value);
+        updateYScale(this.scale.yScale, currentYExtent);
+        updateYAxis(this.axis.yAxis, this.scale.yScale);
+        this.svg.select('.y-axis')
+            .call(this.axis.yAxis);
+
+        //Redraw the current ts
+        select('#ts-current')
+            .attr('d', this.currentLine(this._tsData.current));
+    }
+
```

Comment:
From some reading I've been doing, I think there's a pattern we can use for adding/removing using `selection.enter().append()` to add new elements, them `selection.exit().remove()` to dispose of the old ones.

I think your code looks good as-is, though, it's more of a future thing.

Reply:
Seems like enter() and exit() is useful if the data changed but maybe it can be used to add/remove new things to the graph. Will be interesting to explore.